lighthouse-core/audits/lcp-lazy-loaded.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./audit.js');
+const i18n = require('../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Descriptive title of a diagnostic audit that provides */
+  title: 'LCP element was lazy-loaded',
+  /** Description of a Lighthouse audit that tells */
+  description: '',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class LargestContentfulPaintElement extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'lcp-lazy-loaded',
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      scoreDisplayMode: Audit.SCORING_MODES.ERROR,
+      supportedModes: ['navigation'],
+      requiredArtifacts: ['traces', 'TraceElements', 'ImageElements'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    const images = artifacts.ImageElements;
+    const lazyLoadedImages = images.filter(
+      image => image.loading === 'lazy');
+    const lcpElement = artifacts.TraceElements
+      .find(element => element.traceEventType === 'largest-contentful-paint');
+
+    const lcpElementDetails = [];
+    if (lcpElement) {
+      const lcpImageElement = lazyLoadedImages.find(elem => {
+        return elem.node.devtoolsNodePath === lcpElement.node.devtoolsNodePath;
+      });
+      if (lcpImageElement) {
+        lcpElementDetails.push({
+          node: Audit.makeNodeItem(lcpImageElement.node),
+        });
+      }
+    }
+
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headings = [
+      {key: 'node', itemType: 'node', text: str_(i18n.UIStrings.columnElement)},
+    ];
+
+    const details = Audit.makeTableDetails(headings, lcpElementDetails);
+
+    const displayValue = str_(i18n.UIStrings.displayValueElementsFound,
+      {nodeCount: lcpElementDetails.length});
+
+    return {
+      score: 1,
+      notApplicable: lcpElementDetails.length === 0,
```

Comment:
there are some good edge cases discussed in https://github.com/GoogleChrome/lighthouse/issues/12785 , we probably also want to mark this as notApplicable if the scroll position isn't at the top (in case we measured a URL with a fragment and out-of-viewport anchor)

I don't believe we currently capture this information anywhere in our artifacts, but I think it makes a lot of sense in `ViewportDimensions`, we might discuss in eng sync next week :)

FWIW, our `notApplicable` logic can also be a little confusing so we tend to have an explicit branch for it

```js
if (isAuditNotApplicable()) {
  return {score: 1, notApplicable: true};
}

return {
  score: // the actual results...
}

Reply:
alright we discussed this eng sync and we're fine to proceed here, we just need to account for the not applicable cases:

- gather mode is not `navigation` (you're already doing this 👍 )
- the viewport top is greater than 1/4 of the viewportHeight

Reply:
Great, thanks. I would like to clarify what do you mean viewport top and why greater than 1/4? Would something like this work for you? 
```
        const imageTop = lcpImageElement.clientRect.top;
        const viewportHeight = artifacts.ViewportDimensions.innerHeight;
        if (imageTop / viewportHeight * 100 > 25) {
          lcpElementDetails.push({
            node: Audit.makeNodeItem(lcpImageElement.node),
          });
        }
```

Reply:
Yes that's another way to go about it! Though I think we want a simple `imageTop < viewportHeight` if we're approaching it from the image clientRect angle.

The goal here is the ensure we're marking the audit notApplicable when we're analyzing anything other than the top of the page (LCP element being in the first viewport *technically* could still be wrong if it rendered *before* the scroll, but seems like we can live with that edge case for now)